daal4py/sklearn/neighbors/knn.py

```diff
@@ -161,6 +162,9 @@ class KNeighborsClassifier(BaseKNeighborsClassifier, KNeighborsMixin):
     def predict(self, X):
         X = check_array(X, accept_sparse='csr')
 
+        if hasattr(self, 'n_features_in_') and X.shape[1] != self.n_features_in_:
```

Comment:
Does 'n_features_in_' always exist and have correct value in this class?

Reply:
As I saw in tests it doesn't always exist. That's why I'm using `getattr`